frontend/test/__support__/server-mocks/collection.ts

```diff
@@ -104,46 +104,44 @@ function setupCollectionWithErrorById({
   error: string;
   status?: number;
 }) {
-  fetchMock.get(/api\/collection\/\d+/, () => {
+  fetchMock.get(/api\/collection\/\d+|personal|root/, () => {
     return {
       body: error,
       status,
     };
   });
 }
 
-export function setupRootCollectionItemsEndpoint({
-  collections = [],
-  dashboards = [],
-}: {
-  collections: Collection[];
-  dashboards: Dashboard[];
-}) {
-  fetchMock.get("path:/api/collection/root/items", () => {
-    const rootDashboards = dashboards.filter(
-      dashboard => dashboard.collection_id === null,
-    );
-    const rootCollections = collections.filter(
-      collection => collection.location !== "/",
-    );
-    const data = [...rootDashboards, ...rootCollections];
-
-    return {
-      total: data.length,
-      data,
-    };
-  });
-}
-
 export function setupCollectionItemsEndpoint({
+  collections = [],
   dashboards = [],
 }: {
   dashboards: Dashboard[];
+  collections: Collection[];
 }) {
-  fetchMock.get(/api\/collection\/\d+\/items/, url => {
+  fetchMock.get(/api\/collection\/(\d+|root)\/items/, url => {
     const collectionIdParam = url.split("/")[5];
 
     const collectionId = Number(collectionIdParam);
+    const isRootOrPersonal = Number.isNaN(collectionId);
+
+    // root or personal
+    if (isRootOrPersonal) {
+      if (collectionIdParam === "root") {
```

Comment:
moved logic here, under the single endpoint setup, any idea how to deal with PERSONAL?